tests/cpp/test_dynamic_transform.cpp

```diff
@@ -1370,4 +1373,37 @@ TEST_F(NVFuserTest, DynamicSqueezeTrivialWelford) {
   testValidate(fusion, outputs, {t0}, __LINE__, __FILE__);
 }
 
+TEST_F(DynamicTransformTest, LoopSplit) {
+  const int b = 2, s = 3, h = 96, e = 128;
+
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+  TensorView* in = makeContigConcreteTensor({-1, -1, 12288});
+  TensorView* out = reshape(
+      in,
+      {shape(in)[0],
+       shape(in)[1],
+       IrBuilder::create<Val>(96),
+       IrBuilder::create<Val>(128)});
+  fusion.addInput(in);
+  fusion.addOutput(out);
+
+  const int d = 2;
+  auto mesh = DeviceMesh::createForNumDevices(d);
+  for (auto* tv : {in, out}) {
+    tv->setDeviceMesh(mesh);
+    tv->split(2, d, /*inner_split=*/false);
+    tv->axis(2)->parallelize(ParallelType::DIDx);
+    tv->setAllocationDomain(tv->getLoopDomain(), true);
```

Comment:
out of curiosity, is the reason that we needed a replay on loop, because somewhere else we are expecting to see DIDx on loop as well as on allocation?

My naive mental model sees this as DIDx is specified on allocation domain, with the existing replay, it felt almost like we have some unnecessary redundancy. i.e. loop and allocation need to have some mapping parallel type.